### ISSUE TITLE
the [watchdog timers](https://github.com/DMTF/libspdm/blob/main/os_stub/platform_lib/watchdog.c) currently don't do anything

### ISSUE THREAD

**Issue by uvinrg** [DMTF/libspdm#2419]
the [watchdog timers](https://github.com/DMTF/libspdm/blob/main/os_stub/platform_lib/watchdog.c) currently don't do anything, at least Linux support is needed

_Originally posted by @steven-bellock in https://github.com/DMTF/libspdm/pull/2402#discussion_r1376383768_

**Comment by jyao1** [comment]
That is expected. The platform lib should be customized by every platform.

**Comment by steven-bellock** [comment]
> The platform lib should be customized by every platform.

Yes, and the time has come to customize it for Linux user mode applications.
